corehq/apps/userreports/management/commands/async_rebuild_table.py

```diff
@@ -77,32 +80,36 @@ def _save_ids(self, ids):
                 change = FakeChange(id_, self.fake_change_doc)
                 AsyncIndicator.update_from_kafka_change(change, self.config_ids)
 
-    def _get_ids_to_process(self):
+    def _get_ids_to_process(self, start_date):
         from corehq.sql_db.util import get_db_aliases_for_partitioned_query
         dbs = get_db_aliases_for_partitioned_query()
         if self.database:
             dbs = [db for db in dbs if db == self.database]
         for db in dbs:
-            ids_ = self._get_ids(db)
+            ids_ = self._get_ids(db, start_date)
             num_ids = len(ids_)
             print("processing %d docs from db %s" % (num_ids, db))
             for i, id_ in enumerate(ids_):
                 yield id_
                 if i % 1000 == 0:
                     print("processed %d / %d docs from db %s" % (i, num_ids, db))
 
-    def _get_ids(self, db):
+    def _get_ids(self, db, start_date):
         if self.referenced_type == CASE_DOC_TYPE:
-            return (
+            cases = (
                 CommCareCaseSQL.objects
                 .using(db)
                 .filter(domain=self.domain, type=self.case_type_or_xmlns)
-                .values_list('case_id', flat=True)
             )
+            if start_date:
+                cases = cases.filter(modified_on__gte=start_date)
```

Comment:
```suggestion
                cases = cases.filter(server_modified_on__gte=start_date)
```

Reply:
thanks @snopoke , that column because its indexed or its actually the correct value to use?

Reply:
both

Reply:
i understand the index bit, but why that it is the right column to filter on? would not it be the same value for both columns, unless the don't do mean what they say.

Reply:
`modified_on` is the phone date, `server_modified_on` is when the change got applied on the server

Reply:
oh ho, okay.